src/ScratchViews.C

```diff
@@ -344,44 +414,18 @@ void fill_pre_req_data(
       ThrowRequireMsg(false, "Only node and element fields supported currently.");
     }
   } 
-      
-  SharedMemView<double**>* coordsView = nullptr;
-  if (coordField != nullptr) {
-    coordsView = &prereqData.get_scratch_view_2D(*coordField);
-  }
 
-  const std::set<ELEM_DATA_NEEDED>& dataEnums = dataNeeded.get_data_enums();
-  double error = 0;
-  for(ELEM_DATA_NEEDED data : dataEnums) {
-    switch(data)
-    {
-      case SCS_AREAV:
-         ThrowRequireMsg(meSCS != nullptr, "ERROR, meSCS needs to be non-null if SCS_AREAV is requested.");
-         ThrowRequireMsg(coordsView != nullptr, "ERROR, coords null but SCS_AREAV requested.");
-         meSCS->determinant(1, &((*coordsView)(0,0)), &prereqData.scs_areav(0,0), &error);
-         break;
-      case SCS_GRAD_OP:
-         ThrowRequireMsg(meSCS != nullptr, "ERROR, meSCS needs to be non-null if SCS_GRAD_OP is requested.");
-         ThrowRequireMsg(coordsView != nullptr, "ERROR, coords null but SCS_GRAD_OP requested.");
-         meSCS->grad_op(1, &((*coordsView)(0,0)), &prereqData.dndx(0,0,0), &prereqData.deriv(0), &prereqData.det_j(0), &error);
-         break;
-      case SCS_SHIFTED_GRAD_OP:
-        ThrowRequireMsg(meSCS != nullptr, "ERROR, meSCS needs to be non-null if SCS_GRAD_OP is requested.");
-        ThrowRequireMsg(coordsView != nullptr, "ERROR, coords null but SCS_GRAD_OP requested.");
-        meSCS->shifted_grad_op(1, &((*coordsView)(0,0)), &prereqData.dndx_shifted(0,0,0), &prereqData.deriv(0), &prereqData.det_j(0), &error);
-        break;
-      case SCS_GIJ:
-         ThrowRequireMsg(meSCS != nullptr, "ERROR, meSCS needs to be non-null if SCS_GIJ is requested.");
-         ThrowRequireMsg(coordsView != nullptr, "ERROR, coords null but SCS_GIJ requested.");
-         meSCS->gij(&((*coordsView)(0,0)), &prereqData.gijUpper(0,0,0), &prereqData.gijLower(0,0,0), &prereqData.deriv(0));
-         break;
-      case SCV_VOLUME:
-         ThrowRequireMsg(meSCV != nullptr, "ERROR, meSCV needs to be non-null if SCV_VOLUME is requested.");
-         ThrowRequireMsg(coordsView != nullptr, "ERROR, coords null but SCV_VOLUME requested.");
-         meSCV->determinant(1, &((*coordsView)(0,0)), &prereqData.scv_volume(0), &error);
-         break;
-      default: break;
-    }
+
+  for (auto it = dataNeeded.get_coordinates_map().begin();
```

Comment:
@spdomin So when the user registers coordinates (model or current), we store it in ElemDataRequests. And in the `fill_pre_req_data` call, we loop over the different `coordinates` registered and populate separate `MasterElementViews` for data related to each coordinate type.